src/components/search/results/partial-results.vue

```diff
@@ -1,32 +1,55 @@
 <template>
-  <PartialResultsList>
+  <PartialResultsList class="x-margin--bottom-06 x-list--gap-12">
     <template #default="{ partialResult }">
-      <span>{{ partialResult.query }}</span>
-      <BaseGrid :columns="$x.device === 'mobile' ? 2 : 4" :items="partialResult.results">
+      <div class="x-list x-list--horizontal x-list--align-center x-list--gap-02">
+        <span class="x-title3 x-font-weight--bold">
+          {{
+            $t('partialResults.query', {
+              query: partialResult.query
+            })
+          }}
+        </span>
+        <span class="x-title3 x-font-weight--regular">
+          {{
+            $t('partialResults.totalResults', {
+              totalResults: partialResult.totalResults
+            })
+          }}
+        </span>
+        <PartialQueryButton
+          :query="partialResult.query"
+          class="x-title3 x-font-weight--bold x-button--ghost x-margin--left-auto"
+        >
+          {{ $t('partialResults.viewResults') }}
+          <ChevronRightIcon />
+        </PartialQueryButton>
+      </div>
+      <BaseGrid columns="4" :items="partialResult.results.slice(0, maxItemsToRender)">
         <template #result="{ item }">
           <Result :result="item" />
         </template>
       </BaseGrid>
-      <PartialQueryButton :query="partialResult.query">
-        {{ $t('partialResults.message', partialResult.results.length) }}
-      </PartialQueryButton>
     </template>
   </PartialResultsList>
 </template>
 
 <script lang="ts">
-  import { BaseGrid } from '@empathyco/x-components';
-  import { Component, Vue } from 'vue-property-decorator';
+  import { BaseGrid, ChevronRightIcon } from '@empathyco/x-components';
+  import { Component, Prop, Vue } from 'vue-property-decorator';
   import { PartialQueryButton, PartialResultsList } from '@empathyco/x-components/search';
   import ResultComponent from '../../results/result.vue';
 
   @Component({
     components: {
       BaseGrid,
+      ChevronRightIcon,
       Result: ResultComponent,
       PartialResultsList,
       PartialQueryButton
     }
   })
-  export default class PartialResults extends Vue {}
+  export default class PartialResults extends Vue {
+    @Prop({ default: 4 })
+    protected maxItemsToRender!: number;
```

Comment:
why do we need this prop? as far as I can see it is not even used.

Reply:
Finally I remove it the prop, this should be done in procut